ODIN_II/SRC/ast_loop_unroll.cpp

```diff
@@ -599,109 +393,31 @@ ast_node_t* dup_and_fill_body(ast_node_t *ast_module, ast_node_t* body, ast_node
-					}
-					else
-					{
-						long j;
-						/* find the correct instance in the table */
-						bool found = false;
-						char *instance_name = make_full_ref_name(ast_module->children[0]->types.identifier,
-								child->children[0]->types.identifier,
-								child->children[1]->children[0]->types.identifier,
-								NULL, -1);
-
-						char *temp_instance_name = NULL;
-
-						for (j = 0; !found && j < (*num_original); j++)
-						{
-							// make full ref name of this original 
-							temp_instance_name = make_full_ref_name(ast_module->children[0]->types.identifier,
-								(*instances)[j][0]->children[0]->types.identifier,
-								(*instances)[j][0]->children[1]->children[0]->types.identifier,
-								NULL, -1);
-
-							// if they match, found = true and this_instance = j
-							if (!strcmp(instance_name, temp_instance_name))
-							{
-								found = true;
-								idx = j;
-							}
-
-							vtr::free(temp_instance_name);
-						}
-						oassert(found);
-						vtr::free(instance_name);
-					}
-
-					/* give this unrolled instance a unique name */
-					copy->children[i]->children[1] = replace_named_module(child->children[1], value);
-					oassert(copy->children[i]->children[1]);
-
-					/* find and replace iteration symbol for port connections and parameters */
-					ast_node_t *named_instance = child->children[1];
-					copy->children[i]->children[1] = dup_and_fill_body(ast_module, named_instance, pre, value, error_code, instances, num_unrolled, num_original, is_nested);
-					free_whole_tree(named_instance);
-
-					/* then add it to the table of unrolled instances */
-					(*instances)[idx] = (ast_node_t**)vtr::realloc((*instances)[idx], sizeof(ast_node_t*)*((*num_unrolled)+2));
-					(*instances)[idx][(*num_unrolled)+1] = ast_node_deep_copy(copy->children[i]);
-				}
-				else
-				{
-					/* give this unrolled instance a unique name */
-					copy->children[i]->children[1] = replace_named_module(child->children[1], value);
-					oassert(copy->children[i]->children[1]);
-
-					/* find and replace iteration symbol for port connections and parameters */
-					ast_node_t *named_instance = child->children[1];
-					copy->children[i]->children[1] = dup_and_fill_body(ast_module, named_instance, pre, value, error_code, instances, num_unrolled, num_original, is_nested);
-					free_whole_tree(named_instance);
-				}
+				/* give this unrolled instance a unique name */
+				copy->children[i]->children[1] = replace_named_module(child->children[1], value);
+				oassert(copy->children[i]->children[1]);
+
+				/* find and replace iteration symbol for port connections and parameters */
+				ast_node_t *named_instance = child->children[1];
+				copy->children[i]->children[1] = dup_and_fill_body(ast_module, named_instance, pre, value, error_code);
+				free_whole_tree(named_instance);
 			} 
 
 			if(child && child->num_children > 0)
 			{
 				if (child->type == MODULE_INSTANCE && child->children[0]->type != MODULE_INSTANCE)
```

Comment:
use the said boolean here, and use if(!unrolled) then ... rather then if(unrolled) then do nothing else ...